spring-cloud-stream-integration-tests/src/test/java/org/springframework/cloud/stream/config/StreamListenerHandlerBeanTests.java

```diff
@@ -0,0 +1,140 @@
+import org.springframework.messaging.handler.annotation.SendTo;
+import org.springframework.util.MimeType;
+import org.springframework.util.MimeTypeUtils;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Marius Bogoevici
+ * @author Ilayaperumal Gopinathan
+ */
+@RunWith(Parameterized.class)
+public class StreamListenerHandlerBeanTests {
+
+	private Class<?> configClass;
+
+	public StreamListenerHandlerBeanTests(Class<?> configClass) {
+		this.configClass = configClass;
+	}
+
+	@Parameterized.Parameters
+	public static Collection InputConfigs() {
+		return Arrays.asList(new Class[] { TestHandlerBean1.class, TestHandlerBean2.class });
+	}
+
+	@Test
+	@SuppressWarnings("unchecked")
+	public void testHandlerBean() throws Exception {
+		ConfigurableApplicationContext context = SpringApplication.run(this.configClass,
+				"--spring.cloud.stream.bindings.output.contentType=application/json",
+				"--server.port=0");
+		MessageCollector collector = context.getBean(MessageCollector.class);
+		Processor processor = context.getBean(Processor.class);
+		String id = UUID.randomUUID().toString();
+		processor.input().send(
+				MessageBuilder.withPayload("{\"foo\":\"barbar" + id + "\"}")
+						.setHeader("contentType", "application/json").build());
+		HandlerBean handlerBean = context.getBean(HandlerBean.class);
+		assertThat(handlerBean.receivedPojos).hasSize(1);
+		assertThat(handlerBean.receivedPojos.get(0)).hasFieldOrPropertyWithValue("foo",
+				"barbar" + id);
+		Message<String> message = (Message<String>) collector.forChannel(
+				processor.output()).poll(1, TimeUnit.SECONDS);
+		assertThat(message).isNotNull();
+		assertThat(message.getPayload()).isEqualTo("{\"bar\":\"barbar" + id + "\"}");
+		assertThat(message.getHeaders().get(MessageHeaders.CONTENT_TYPE, MimeType.class)
+				.includes(MimeTypeUtils.APPLICATION_JSON));
+		context.close();
+	}
+
+	@EnableBinding(Processor.class)
+	@EnableAutoConfiguration
+	public static class TestHandlerBean1 {
+
+		@Bean
+		public HandlerBean1 handlerBean() {
+			return new HandlerBean1();
+		}
+	}
+
+	@EnableBinding(Processor.class)
+	@EnableAutoConfiguration
+	public static class TestHandlerBean2 {
+
+		@Bean
+		public HandlerBean2 handlerBean() {
+			return new HandlerBean2();
+		}
+	}
+
+	public static class HandlerBean1 extends HandlerBean {
+
+		@StreamListener(Processor.INPUT)
+		@SendTo(Processor.OUTPUT)
+		public StreamListenerTestInterfaces.BarPojo receive(StreamListenerTestInterfaces.FooPojo fooMessage) {
+			this.receivedPojos.add(fooMessage);
+			StreamListenerTestInterfaces.BarPojo barPojo = new StreamListenerTestInterfaces.BarPojo();
+			barPojo.setBar(fooMessage.getFoo());
+			return barPojo;
+		}
+	}
+
+	public static class HandlerBean2 extends HandlerBean {
+
+		@StreamListener(Processor.INPUT)
+		@Output(Processor.OUTPUT)
+		public StreamListenerTestInterfaces.BarPojo receive(StreamListenerTestInterfaces.FooPojo fooMessage) {
```

Comment:
We should support `@SendTo`  for both bindable and POJO/Message but `@Output` only for bindable types.